Vue3: Additional install/upgrade chart steps fail to load

<!--------- For bugs and general issues --------->
**Setup**
- Rancher version: dev if 2.10

**Describe the bug**
- Charts that have built-in install experiences can contain conditional additional steps
- these are defined as vue components in `shell/chart/<chart name/steps`
- this is a process brought in with https://github.com/rancher/dashboard/pull/3101, however the actual steps have been removed (v1 monitoring no longer supported)
- whilst testing https://github.com/rancher/dashboard/issues/11773 found out the underlying process is broken 


> Note 1: **broken feature isn't currently being used**
> Note 2: I tagged this with vue3, but haven't confirmed working in vue2

**To Reproduce**
- Add the following to a new file at `shell/chart/monitoring/steps/my-step.vue`
  - ```
    <script>
    export default {
      label:      'generic.add',
      subtext:    'generic.back',
      weight:     100,
      components: {},
      data() {
        return {};
      },
      mounted() {
        this.$emit('update', {
          loading: false,
          ready:   true,
          hidden:  false,
        });
      },
    };
    </script>
    <template>
      <div>
        Hello World
      </div>
    </template>
    ```
- Run the dashboard
- Nav to Apps --> Charts --> Monitoring --> Install
 
**Result**

```
fetch.client.js:75 Error in fetch(): TypeError: customStep.component is not a function
    at Proxy.loadChartStep (install.vue:934:1)
    at install.vue:929:1
    at Array.map (<anonymous>)
    at Proxy.loadChartSteps (install.vue:929:1)
    at async Proxy.fetch (install.vue:269:1)
    at async Proxy.$_fetch (fetch.client.js:72:1)
```


**Expected Result**
- Additional step show in install chart wizard

**Additional context**
- see `shell/store/catalog.js `chartSteps`, `shell/pages/c/_cluster/apps/charts/install.vue` `loadChartSteps`

--- a/shell/pages/c/_cluster/apps/charts/install.vue
+++ b/shell/pages/c/_cluster/apps/charts/install.vue
@@ -5,6 +5,7 @@ import isEqual from 'lodash/isEqual';
 import { mapPref, DIFF } from '@shell/store/prefs';
 import { mapFeature, MULTI_CLUSTER, LEGACY } from '@shell/store/features';
 import { mapGetters } from 'vuex';
+import { markRaw } from 'vue';
 import { Banner } from '@components/Banner';
 import ButtonGroup from '@shell/components/ButtonGroup';
 import ChartReadme from '@shell/components/ChartReadme';
@@ -893,7 +894,7 @@ export default {
         const hasChartComponent = this.$store.getters['type-map/hasCustomChart'](component);
 
         if ( hasChartComponent ) {
-          this.valuesComponent = this.$store.getters['type-map/importChart'](component);
+          this.valuesComponent = markRaw(this.$store.getters['type-map/importChart'](component));
           this.showValuesComponent = true;
         } else {
           this.valuesComponent = null;

--- a/shell/pages/c/_cluster/apps/charts/install.vue
+++ b/shell/pages/c/_cluster/apps/charts/install.vue
@@ -261,14 +261,6 @@ export default {
     }
 
     /*
-      Check if the Helm chart has indicated
-      that the user should fill out the chart values
-      through a wizard-style workflow. If so, load
-      the chart steps.
-    */
-    await this.loadChartSteps();
-
-    /*
       this.loadedVersion will only be true if you select a non-defalut
       option from the "Version" dropdown menu in Apps & Marketplace
       when updating a previously installed app.
@@ -450,10 +442,6 @@ export default {
         ready:          true,
         weight:         10
       },
-
-      customSteps: [
-
-      ],
 
       isPlainLayout: isPlainLayout(this.$route.query),
 
@@ -674,7 +662,6 @@ export default {
         steps.push(
           this.stepBasic,
           this.stepValues,
-          ...this.customSteps
         );
       }
 
@@ -826,10 +813,6 @@ export default {
     // for editing values
     await this.loadValuesComponent();
 
-    // Load Helm chart info used for showing
-    // wizard steps
-    await this.loadChartSteps();
-
     window.scrollTop = 0;
 
     this.preFormYamlOption = this.valuesComponent || this.hasQuestions ? VALUES_STATE.FORM : VALUES_STATE.YAML;
@@ -920,33 +903,6 @@ export default {
         this.valuesComponent = null;
         this.showValuesComponent = false;
       }
-    },
-
-    async loadChartSteps() {
-      const component = this.version?.annotations?.[CATALOG_ANNOTATIONS.COMPONENT];
-
-      if ( component ) {
-        const steps = await this.$store.getters['catalog/chartSteps'](component);
-
-        this.customSteps = await Promise.all( steps.map((cs) => this.loadChartStep(cs)));
-      }
-    },
-
-    async loadChartStep(customStep) {
-      // Broken in 2.10, see https://github.com/rancher/dashboard/issues/11898
-      const loaded = await customStep.component();
-      const withFallBack = this.$store.getters['i18n/withFallback'];
-
-      return {
-        name:      customStep.name,
-        label:     withFallBack(loaded?.default?.label, null, customStep.name),
-        subtext:   withFallBack(loaded?.default?.subtext, null, ''),
-        weight:    loaded?.default?.weight,
-        ready:     false,
-        hidden:    true,
-        loading:   true,
-        component: customStep.component,
-      };
     },
 
     selectChart(chart) {
@@ -1348,17 +1304,6 @@ export default {
       @cancel="cancel"
       @finish="finish"
     >
-      <template
-        v-for="customStep of customSteps"
-        v-slot:[customStep.name]
-        :key="customStep.name"
-      >
-        <component
-          :is="customStep.component"
-          @update="updateStep(customStep.name, $event)"
-          @errors="e=>errors.push(...e)"
-        />
-      </template>
       <template #bannerTitleImage>
         <div>
           <div class="logo-bg">

--- a/shell/store/catalog.js
+++ b/shell/store/catalog.js
@@ -7,7 +7,6 @@ import { findBy, addObject, filterBy, isArray } from '@shell/utils/array';
 import { stringify } from '@shell/utils/error';
 import { classify } from '@shell/plugins/dashboard-store/classify';
 import { sortBy } from '@shell/utils/sort';
-import { importChart } from '@shell/utils/dynamic-importer';
 import { ensureRegex } from '@shell/utils/string';
 import { isPrerelease } from '@shell/utils/version';
 import difference from 'lodash/difference';
@@ -262,31 +261,6 @@ export const getters = {
   importComponent(state, getters) {
     return (name) => {
       return getters['type-map/importChart'](name);
-    };
-  },
-
-  chartSteps(state, getters) {
-    return (name) => {
-      const steps = [];
-
-      const stepsPath = `./${ name }/steps/`;
-      // require.context only takes literals, so find all candidate step files and filter out
-      const allPaths = require.context('@shell/chart', true, /\.vue$/).keys();
-
-      allPaths
-        .filter((path) => path.startsWith(stepsPath))
-        .forEach((path) => {
-          try {
-            steps.push({
-              name:      path.replace(stepsPath, ''),
-              component: importChart(path.substr(2, path.length)),
-            });
-          } catch (e) {
-            console.warn(`Failed to load step component ${ path } for chart ${ name }`, e); // eslint-disable-line no-console
-          }
-        });
-
-      return steps;
     };
   },
 

